includes/handlers/class-webmention-handler-jsonld.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Class for webmention parsing using JSON-LD.
+ */
+class Webmention_Handler_JSONLD extends Webmention_Handler_Base {
+
+	/**
+	 * Handler Slug to Uniquely Identify it.
+	 *
+	 * @var string
+	 */
+	protected $slug = 'jsonld';
+
+	/**
+	 * Takes a request object and parses it.
+	 *
+	 * @param Webmention_Request $request Request Object.
+	 * @param Webmention_Item $item A Parsed Item. If null, a new one will be created.
+	 * @return WP_Error|true Return error or true if successful.
+	 */
+	public function parse( $request, $item = null ) {
+		if ( $item instanceof Webmention_Item ) {
+			$this->set_webmention_item( $item );
+		}
+
+		$dom   = clone $request->get_domdocument();
+		$xpath = new DOMXPath( $dom );
+
+		$jsonld  = array();
+		$content = '';
+		foreach ( $xpath->query( "//script[@type='application/ld+json']" ) as $script ) {
+			$content  = $script->textContent; // phpcs:ignore
+			$jsonld[] = json_decode( $content, true );
+		}
+		$jsonld = array_filter( $jsonld );
+		if ( 1 === count( $jsonld ) && wp_is_numeric_array( $jsonld[0] ) ) {
+			$jsonld = $jsonld[0];
+		}
+
+		// Set raw data.
+		$this->webmention_item->set__raw( $jsonld );
+		$this->webmention_item->set__response_type = 'mention';
+
+		foreach ( $jsonld as $json ) {
+			if ( ! $this->is_jsonld( $json ) ) {
+				continue;
+			}
+			if ( in_array( $json['@type'], array( 'WebPage', 'Article', 'NewsArticle' ), true ) ) {
+				if ( isset( $json['datePublished'] ) ) {
+					$this->webmention_item->set_published( new DateTimeImmutable( $json['datePublished'] ) );
+				}
+				if ( isset( $json['dateModified'] ) ) {
+					$this->webmention_item->set_updated( new DateTimeImmutable( $json['dateModified'] ) );
+				}
+
+				$this->webmention_item->set_name( $json['headline'] );
+				$this->webmention_item->set_name( $json['name'] );
+				$this->webmention_item->set_summary( $json['description'] );
+				$this->webmention_item->set_category( $json['keywords'] );
+
+				if ( isset( $json['articleBody'] ) ) {
+					$html            = webmention_sanitize_html( $json['articleBody'] );
+					$json['content'] = array(
```

Comment:
we defined `$content` as string.

Reply:
Sorry about that. Was already thinking MF2